build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/MrjarPlugin.java

```diff
@@ -75,7 +75,11 @@ public void apply(Project project) {
             String testSourceSetName = SourceSet.TEST_SOURCE_SET_NAME + javaVersion;
             SourceSet testSourceSet = addSourceSet(project, javaExtension, testSourceSetName, testSourceSets, javaVersion);
             testSourceSets.add(testSourceSetName);
-            createTestTask(project, testSourceSet, javaVersion, mainSourceSets);
+            var testTask = project.getTasks().withType(Test.class).iterator().next();
```

Comment:
I believe this is forcing the test tasks to be realized at configuration time, which we don't want.